libs/blockchain-watcher/feature-app/src/lib/Klass/PaymentCreatedEvent.ts

```diff
@@ -0,0 +1,69 @@
+import { EnumPaymentCreatedEventType } from '../Enum/EnumPaymentCreatedEventType'
+import { v4 } from 'uuid'
+import {
+  IsEnum,
+  IsNotIn,
+  IsNumber,
+  IsString,
+  IsUUID,
+  Max,
+  Min,
+  validateSync,
+} from 'class-validator'
+import { IsEthAddress } from '../util/IsEthAddress'
+
+export class PaymentCreatedEvent {
+  @IsUUID()
+  @IsString()
+  id: string = v4()
+
+  @IsNotIn([EnumPaymentCreatedEventType.unknown])
+  @IsEnum(EnumPaymentCreatedEventType)
+  type: EnumPaymentCreatedEventType = EnumPaymentCreatedEventType.unknown
+
+  @IsEthAddress()
+  @IsString()
+  party1: string = ''
+
+  @IsEthAddress()
+  @IsString()
+  party2: string = ''
+
+  @IsEthAddress()
+  @IsString()
+  contract: string = ''
+
+  @IsEthAddress()
+  @IsString()
+  judgeContract: string = ''
+
+  @IsEthAddress()
+  @IsString()
+  token: string = ''
+
+  @Min(0)
+  @Max(100)
+  @IsNumber()
+  discount: number = 0
+
+  @Min(0)
+  @IsNumber()
+  ethPaid: number = 0
+
+  isValid() {
+    let errArr = validateSync(this)
+
+    if (errArr.length > 0) {
+      for (let err of errArr) {
+        console.error(err)
+      }
+      return false
+    }
+
+    return true
+  }
+
+  constructor(props: Partial<PaymentCreatedEvent>) {
+    Object.assign(this, props)
```

Comment:
Could it be useful to use `class-transformer` instead?

Reply:
I'll keep the `class-transformer` in mind, but I don't think it's necessary in this case. `Partial<Klass>` and `Object.assign()` works fine and clean, and they are native code.

Reply:
Ok 👌